src/v/kafka/protocol/schemata/generator.py

```diff
@@ -1128,6 +1148,51 @@ def is_error_code(self):
 {%- endif %}
 {% endmacro %}
 
+{% macro render_formatter(struct) %}
+{%- if not struct.is_streamable %}
+template<>
+struct fmt::formatter<kafka::{{ struct.name }}, char> {
+
+    template <typename ParseContext>
+    constexpr auto parse(ParseContext &ctx) -> decltype(ctx.begin()) {
+        return ctx.begin();
+    }
```

Comment:
I'd be tempted to extract the parsing as a base class so that it's not a separate instantiation for every type, and will make it easier to extend in the future. A reasonable base class might already exist as `fmt::formatter<std::string_view>`.
```suggestion
using formatter_base = fmt::formatter<std::string_view>;

template<>
struct fmt::formatter<kafka::{{ struct.name }}, char> : formatter_base {
```
You can probably come up with a way to not have that in the header.

An example of a useful base class might be to provide a limit that's configurable as a cluster config, unless specified in the format string. (not suggesting this functionality for now, just throwing out ideas).

Reply:
> provide a limit that's configurable as a cluster config

cool idea for sure